ygen/gogen.go

```diff
@@ -502,6 +502,59 @@ func (t *{{ .Receiver }}) New{{ .ListName }}(
 
 	return t.{{ .ListName }}[key], nil
 }
+`
+
+	// goListMemberRenameTemplate provides a template for a function which renames
+	// an entry within a list. It is used to generate functions for each list within
+	// a generated Go struct.
+	goListMemberRenameTemplate = `
+// Rename{{ .ListName }} renames an entry in the list {{ .ListName }} within
+// the {{ .Receiver }} struct. The entry with key oldk is renamed to newk updating
+// the key within the value.
+func (t *{{ .Receiver }}) Rename{{ .ListName }}(
+	{{- if ne .KeyStruct "" -}}
+	oldk, newk {{ .KeyStruct -}}
+  {{- else -}}
+	{{- range $key := .Keys -}}
+	oldk, newk {{ $key.Type -}}
+	{{- end -}}
+	{{- end -}}
+) error {
+	e, ok := t.{{ .ListName }}[oldk]
+	if !ok {
+		return fmt.Errorf("key %v not found in {{ .ListName }}", oldk)
+	}
+
+	ns, err := ygot.DeepCopy(e)
```

Comment:
Does this have to perform a copy?

We could as well get a ref to the existing value in the map, change it's name, delete and re-insert the same object.